cou/utils/juju_utils.py

```diff
@@ -142,13 +142,30 @@ class Machine:
     az: Optional[str] = None  # simple deployments may not have azs
 
 
+@dataclass(frozen=True)
+class SubordinateUnit:
+    """Representation of a single unit of subordinate unit."""
+
+    name: str
+    workload_version: str
```

Comment:
I'm not seeing any use for workload_version, so let's not add it if not necessary.

Reply:
I would like to keep this one, it's like a basic information of the unit.

Reply:
A lot of subordinate apps doesn't even have workload_version and we are determining the OpenStack release by the channel. It's not necessary and most likely will never be used

Reply:
For me it's quiet weird that those two different unit class have two different attributes.

Here I will delete the workload_version since you persist in. But I will say the design of the dataclass in juju_utils have is wrong.

Reply:
Removed.

Reply:
> For me it's quiet weird that those two different unit class have two different attributes.

Juju itself treat totally different subordinate units from principal units. If you access the status of a subordinate unit, it will say that there isn't units. When you access by model.applications it will have units, but it won't have workload_version. Now you showed that by accessing this way potentially show workload_version.

If you want I'm ok to inherit from the Unit or use Unit itself and populate with all the fields, but it might be more complex.